pkg/agent/qrm-plugins/cpu/dynamicpolicy/policy_hint_handlers.go

```diff
@@ -48,19 +48,51 @@ func (p *DynamicPolicy) sharedCoresHintHandler(ctx context.Context,
 		return nil, fmt.Errorf("got nil request")
 	}
 
-	if !qosutil.AnnotationsIndicateNUMABinding(req.Annotations) {
-		return util.PackResourceHintsResponse(req, string(v1.ResourceCPU),
-			map[string]*pluginapi.ListOfTopologyHints{
-				string(v1.ResourceCPU): nil, // indicates that there is no numa preference
-			})
+	if qosutil.AnnotationsIndicateNUMABinding(req.Annotations) {
+		return p.sharedCoresWithNUMABindingHintHandler(ctx, req)
 	}
 
-	return p.sharedCoresWithNUMABindingHintHandler(ctx, req)
+	ok, err := p.checkNormalShareCoresCpuResource(req)
+	if err != nil {
+		general.Errorf("failed to check share cores cpu resource for pod: %s/%s, container: %s",
+			req.PodNamespace, req.PodName, req.ContainerName)
+		return nil, fmt.Errorf("failed to check share cores cpu resource: %q", err)
+	}
+
+	if !ok {
+		_ = p.emitter.StoreInt64(util.MetricNameShareCoresNoEnoughResourceFailed, 1, metrics.MetricTypeNameCount, []metrics.MetricTag{
+			{
+				Key: "resource",
+				Val: "cpu",
+			},
+			{
```

Comment:
make it consistent with bellow? 

<img width="684" alt="image" src="https://github.com/user-attachments/assets/e27667b0-e337-4e81-af50-6902c45fbce1">